application/src/main/java/org/opentripplanner/updater/vehicle_rental/datasources/GbfsFreeVehicleStatusMapper.java

```diff
@@ -54,12 +60,28 @@ public VehicleRentalVehicle mapFreeVehicleStatus(GBFSBike vehicle) {
         vehicle.getLastReported() != null
           ? Instant.ofEpochSecond((long) (double) vehicle.getLastReported())
           : null;
-      RentalVehicleFuel fuel = new RentalVehicleFuel(
-        vehicle.getCurrentFuelPercent(),
-        vehicle.getCurrentRangeMeters() != null
-          ? Distance.ofMeters(vehicle.getCurrentRangeMeters())
-          : null
-      );
+      Double fuelPercent = null;
+      if (vehicle.getCurrentFuelPercent() != null) {
+        try {
+          fuelPercent =
+            (Double) GraphQLScalars.RATIO_SCALAR
```

Comment:
Don't reuse this API code inside this core model. Just write some small validation method, for instance here. One option would also be to introduce some Ratio or whatever class inside the core model which does the validation on construct.

Reply:
We decided we can ignore optional values but we shouldn't correct values (i.e. try to round numbers to an allowed value, for example). If a required value is incorrect, then we should probably just not include the entity (station, floating vehicle etc.) at all.

Reply:
Ok, it's a good method. So in this case I'll exclude the entity only if the rental vehicle has an engine and the current_range_meters is invalid or null.

> Don't reuse this API code inside this core model. Just write some small validation method, for instance here. One option would also be to introduce some Ratio or whatever class inside the core model which does the validation on construct.

I'll go with the Ratio class. It's in line with the Distance class.